backend/schema/room.py

```diff
@@ -78,35 +78,69 @@ def validate_dm(cls, values):
             HTTPException [400]: if DM has topic
             HTTPException [400]: if DM has a description
         """
-        plugin_name = values.get("plugin_name")
-        room_members = values.get("room_members", {})
-        topic = values.get("topic")
-        description = values.get("description")
-        created_by = values.get("created_by")
-
-        if plugin_name == Plugin.DM and len(list(room_members.keys())) > 9:
-            raise HTTPException(
-                status_code=400, detail="Group DM cannot have more than 9 members"
-            )
-
-        if plugin_name == Plugin.DM and topic is not None:
-            raise HTTPException(status_code=400, detail="DM should not have a topic")
-
-        if plugin_name == Plugin.DM and description is not None:
-            raise HTTPException(
-                status_code=400, detail="DM should not have a description"
-            )
-
-        if plugin_name == Plugin.DM.value and created_by is not None:
-            raise HTTPException(
-                status_code=400, detail="DM should not have a created by"
-            )
+        if values["room_type"] != RoomType.CHANNEL:
+            room_type = values.get("room_type")
+            room_members = values.get("room_members", {})
+            topic = values.get("topic")
+            description = values.get("description")
+            org_id = values.get("org_id")
+
+            with concurrent.futures.ThreadPoolExecutor(1) as executor:
+                future = executor.submit(
+                    asyncio.run, get_org_rooms(org_id=org_id, room_type=room_type)
+                )
+                rooms = future.result()
+
+            if rooms is None:  # check if connection is avaliable
+                raise HTTPException(
+                    status_code=status.HTTP_424_FAILED_DEPENDENCY,
+                    detail="unable to read database",
+                )
+
+            if (
+                room_type == RoomType.GROUP_DM
+                and len(set(room_members.keys())) > 9
+                and len(set(room_members.keys())) < 2
+            ):
```

Comment:
We could rewrite this part to simplify the chained comparison:
```suggestion
            if (
                room_type == RoomType.GROUP_DM
                and 9 < len(set(room_members.keys())) < 2
            ):
```
But, as we can see clearly now, the second operand can never be fulfilled. 
What do you think about splitting the second operand with an `OR` operator?

Reply:
Well a group DM is a DM of not less than 3 members and not more than 9 members.
Yea I've chained it with an OR operator. Thanks

Reply:
Thanks.